tests/unit/profiles-management/pmr/test_classes.py

```diff
@@ -0,0 +1,107 @@
+import pytest
+from pydantic import ValidationError
+
+from profiles_management.pmr.classes import (
+    Owner,
+    Profile,
+    ProfilesManagementRepresentation,
+    ResourceQuotaSpecModel,
+    UserKind,
+)
+
+
+@pytest.mark.parametrize(
+    "quota",
+    [
+        {"kimchi": 1},  # arbitrary fields
+        {"scopes": 1},  # incorrect type
+        {"hard": "1000"},  # incorrect type, no keys
+        {"kimchi": 1, "hard": {"cpu": "1000"}},  # arbitrary field plus correct field
+    ],
+)
+def test_invalid_resource_quota(quota):
+    try:
+        Profile(
+            name="test",
+            contributors=[],
+            resources=ResourceQuotaSpecModel.model_validate(quota),
+            owner=Owner(name="kimchi", kind=UserKind.USER),
+        )
+        assert False
+    except ValidationError:
+        assert True
```

Comment:
You should use a context manager here to catch the exception:

```python
with pytest.raises(<exception>) as e:
  Profile(...)
assert <your assertion>
```

Reply:
That's awesome, didn't know about it. Thanks!

c51d187